src/main/java/com/atlassian/db/replica/spi/ReplicaConsistency.java

```diff
@@ -12,12 +12,24 @@
 public interface ReplicaConsistency {
 
     /**
-     * Informs that {@code main} received an UPDATE, INSERT or DELETE.
+     * Informs that {@code main} received an UPDATE, INSERT or DELETE or transaction commit
+     * when in a transaction.
      *
      * @param main connects to the main database
      */
     void write(Connection main);
 
+    /**
+     * Invoked just before transaction commit.
+     *
+     * Notice: The method will not handle all writes. Writes done outside of a transaction
```

Comment:
> The method will not handle all writes

have you thought about adding a test/any other check to check which will be handled, which not?

Reply:
I thought about copy-pasting our current `ReplicaConsistency` implementation from the product and adding IT tests. I think I covered all the cases with unit tests.